### ISSUE TITLE
LTTng warnings in corefx CI runs

### ISSUE THREAD

**Issue by stephentoub** [dotnet/coreclr#1837]
We're seeing a ton of warnings now like this in our corefx CI runs and it's actually breaking some of our tests:

libust[14499/14499]: Warning: HOME environment variable not set. Disabling LTTng-UST per-user tracing. (in setup_local_apps() at lttng-ust-comm.c:305)

What can be done to silence it? This is an issue for more than just our tests: apps shouldn't have to see such a warning message (and lots of them).

cc: @brianrob, @mmitche

**Comment by stephentoub** [comment]
(Note that if I go to the script console for one of our CI machines and execute ```println System.getenv("HOME")```, it does come back as ```null```.)

**Comment by gkhanna79** [comment]
cc @ramarag

**Comment by gkhanna79** [comment]
Here is an example of similar issues others have faced - http://tracker.ceph.com/issues/13470

@brianrob - is this due to PAL initialization of the LTTng SO?

**Comment by brianrob** [comment]
I am looking into this.

@stephentoub, are you seeing this on all flavors of Linux that we run in the CI or just a particular flavor?

**Comment by stephentoub** [comment]
Thanks, Brian.  We only run our corefx tests on Ubuntu 14.04 as part of PRs right now, so that's the only place I've noticed it.

**Comment by stephentoub** [comment]
For reference, here's an example log, with the warning appearing 243 times:
http://dotnet-ci.cloudapp.net/job/dotnet_corefx_linux_debug_tst_prtest/3741/console

**Comment by brianrob** [comment]
Ok, thanks.  Let me do some more research on this.  It appears to be debug logging that should be off by default unless the package is compiled with debug flags.  It's possible that the package was compiled with debug flags - I'm going to try and find out.

**Comment by mmitche** [comment]
I wonder if the lack of HOME might be related to the fact that these are non-interactive sessions on Linux.  For instance if I log on to that machine, it has a HOME env var.

**Comment by brianrob** [comment]
It looks like the issue is that the Ubuntu package for LTTng-UST is compiled with debug logging always on.  This message just happens to be shown because there is no HOME environment variable.

I'm going to follow-up on this with the LTTng maintainers, but in the meantime, we can unblock ourselves by setting the LTTNG_HOME environment variable.  I've already done with with guidance from @mmitche, and have a PR posted to re-enable the tests - https://github.com/dotnet/corefx/pull/4084.

For customers, there are a few choices, but know that these are only applicable for cases where the environment doesn't have a HOME variable.  Anything interactive shouldn't hit this problem.  First, if LTTng-UST isn't installed, this text won't appear, so uninstalling it or not installing it is one option, which is OK if you don't need to do performance data collection or collect tracing data.  If you do need these capabilities, then setting the HOME or LTTNG_HOME variable is an option as well.  If this isn't do-able, it's also possible to install a LTTng-UST package that hasn't been compiled with debug logging on by default, or at worst, compile from source.

**Comment by brianrob** [comment]
This has been fixed in LTTng 2.7 and worked around in the CI.
